src/test/java/bio/terra/service/filedata/google/firestore/FileOperationTest.java

```diff
@@ -255,21 +273,62 @@ public void fileOperationsTest() throws Exception {
             containsString("gs path"));
     }
 
+    // ------ Retry shared lock/unlock tests ---------------
+
     @Test
     public void retryAndAcquireSharedLock() throws Exception {
         FileLoadModel fileLoadModel = makeFileLoad(profileModel.getId());
 
         connectedOperations.retryAcquireLockIngestFileSuccess(
-            true,
+            ConnectedOperations.RetryType.lock, true, true,
+            ConfigEnum.FILE_INGEST_SHARED_LOCK_RETRY_FAULT,
+            datasetSummary.getId(), fileLoadModel, configService, datasetDao);
+    }
+
+    @Test
+    public void retryAndAcquireSharedUnlock() throws Exception {
+        FileLoadModel fileLoadModel = makeFileLoad(profileModel.getId());
+
+        connectedOperations.retryAcquireLockIngestFileSuccess(
+            ConnectedOperations.RetryType.unlock, true,  true,
+            ConfigEnum.FILE_INGEST_SHARED_UNLOCK_RETRY_FAULT,
+            datasetSummary.getId(), fileLoadModel, configService, datasetDao);
+    }
+
+    // These tests can be used as one offs to see if fatal errors are working as expected
+    // But, as is, they shouldn't be run every time because it can leave tests in a bad state
+    // (leftover artifacts after tests)
+    @Ignore
+    @Test
+    public void retryAndEventuallyFailToAcquireSharedLock() throws Exception {
+        FileLoadModel fileLoadModel = makeFileLoad(profileModel.getId());
+
+        connectedOperations.retryAcquireLockIngestFileSuccess(
+            ConnectedOperations.RetryType.lock, false, false,
+            ConfigEnum.FILE_INGEST_SHARED_LOCK_RETRY_FAULT,
+            datasetSummary.getId(), fileLoadModel, configService, datasetDao);
+        configService.setFault(ConfigEnum.FILE_INGEST_SHARED_LOCK_RETRY_FAULT.toString(), false);
+    }
+
+    @Ignore
+    @Test
+    public void retryAndFailAcquireSharedUnlock() throws Exception {
+        FileLoadModel fileLoadModel = makeFileLoad(profileModel.getId());
+
+        connectedOperations.retryAcquireLockIngestFileSuccess(
+            ConnectedOperations.RetryType.unlock, false, true,
+            ConfigEnum.FILE_INGEST_SHARED_UNLOCK_FATAL_FAULT,
```

Comment:
I'm not sure if I should delete these tests. If I do that, I can also slightly simplify the retryAcquireLockIngestFile method.